File: frontend/src/types/comparisonMatrix.ts
export interface MatrixValue {
  summary: string | string[] | null
  claim_ids: number[]
  evidence_ids: number[]
  confidence: number
}

export interface MatrixRow {
  key: string
  label: string
  values: Record<string, MatrixValue>
}

export interface ComparisonMatrix {
  id: number
  task_id: number
  template_key?: string | null
  matrix_type: string
  title: string
  matrix_schema_json: {
    template_key?: string | null
    columns: string[]
    rows: Array<{ key: string; label: string }>
  }
  matrix_data_json: {
    rows: MatrixRow[]
  }
  claim_ids_json?: number[] | null
  evidence_ids_json?: number[] | null
  created_at: string
  updated_at: string
}

File: backend/app/models/comparison_matrix.py
from datetime import datetime

from sqlalchemy import BigInteger, DateTime, Index, JSON, String
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base
from app.core.timezone import now_bj


class ComparisonMatrix(Base):
    __tablename__ = "comparison_matrix"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    task_id: Mapped[int] = mapped_column(BigInteger, nullable=False, index=True)
    template_key: Mapped[str | None] = mapped_column(String(100), index=True)
    matrix_type: Mapped[str] = mapped_column(String(100), nullable=False, index=True)
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    matrix_schema_json: Mapped[dict] = mapped_column(JSON, nullable=False)
    matrix_data_json: Mapped[dict] = mapped_column(JSON, nullable=False)
    claim_ids_json: Mapped[list[int] | None] = mapped_column(JSON)
    evidence_ids_json: Mapped[list[int] | None] = mapped_column(JSON)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=now_bj, nullable=False)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=now_bj, onupdate=now_bj, nullable=False)

    __table_args__ = (Index("ix_matrix_task_type", "task_id", "matrix_type"),)
